lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -1116,6 +1116,33 @@ private function getParameterType(ReflectionParameter $parameter)
         return $this->formatType($parameter->getType(), $declaringFunction, $parameter);
     }
 
+    /**
+     * @return string
+     */
+    private function getParameterDefaultValue(ReflectionParameter $parameter)
+    {
+        if (! $parameter->isDefaultValueAvailable()) {
+            return '';
+        }
+
+        if (PHP_VERSION_ID < 80100) {
+            return ' = ' . var_export($parameter->getDefaultValue(), true);
+        }
+
+        $value = rtrim(substr(explode('$' . $parameter->getName() . ' = ', (string) $parameter, 2)[1], 0, -2));
+        $parts = preg_split('{(\'(?:[^\'\\\\]*(?:\\\\.[^\'\\\\]*)*)\')}', $value, -1, PREG_SPLIT_DELIM_CAPTURE);
+
+        foreach ($parts as $i => &$part) {
+            if ($part === '' || $i % 2 !== 0) {
+                continue;
+            }
+
+            $part = preg_replace('/(?(DEFINE)(?<V>[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+))(?<!\\\\)(?&V)(?:\\\\(?&V))*+::/', '\\\\$0', $part);
```

Comment:
What is the purpose of this line? When I comment it out, the test below does not fail :thinking: … I'm using PHP 8.1.8 Likewise, if I put rubbish in the regex line 1133, no test fails.

Reply:
I have absolutely no clue, I took what @nicolas-grekas provided for ProxyManager for granted 🙈 Tried to go for original FriendsOfPHP PR but nobody challenged that there either

Reply:
Then let's just return what's on line 1132? It seems to be the import part of this patch. When somebody reports the bug the bottom part fixes, let's make sure to add a test that proves it useful.

Reply:
This regexp is supposed to add the `\` prefix to internal classes when they declare a default value that references a class constant.

Reply:
This sounds like it would also affect top-level classes that are not internal, but user defined, wouldn't it? In both cases, I'm unsure if `doctrine/common` should support it, as AFAIK it's only ever used to proxify entities and documents, that should never be in the top-level namespace, or inherit from a class defined in it. Unlike proxy-manager I suppose.

Reply:
Use defined classes always have the backslash prefix that's why only internal ones are concerned.
Agree about entities, unless they could extend et ArrayObject or similar? Not sure it'd make sense.

Reply:
That would be quite weird but I not completely impossible I suppose. @malarzm , I think we should either add a test case based on `ArrayObject::asort()` (which refers to a constant), or drop the code.

Reply:
Added part with using global const, it does fail without those regexes :)

Reply:
Really? When I try again what I did in https://github.com/doctrine/common/pull/966#discussion_r947094561 on eb0aaa523902d77fcc8cbccf06fa7f4f6fd869f3, the test still passes :thinking:

Reply:
> Really?

Dang, sorry. I made an early return in `getDefaultValue` that was supposed to break the test... it did, but for totally other reason :s

Reply:
Sadly, I'm giving up. I'm almost starting my vacations and I think it'll be better to have main cases covered. If somebody reports a bug that could be fixed by regexes I've just removed then it can be done in a bugfix manner.